patches/server/0840-Schoolable-Fish-API.patch

```diff
@@ -0,0 +1,110 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Tue, 7 Dec 2021 19:50:33 -0500
+Subject: [PATCH] Schoolable Fish API
+
+
+diff --git a/src/main/java/io/papermc/paper/entity/PaperSchoolingFish.java b/src/main/java/io/papermc/paper/entity/PaperSchoolingFish.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..6f01e6ad55f98edd114cfd818678e64eac9c3c68
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/entity/PaperSchoolingFish.java
+@@ -0,0 +1,59 @@
++package io.papermc.paper.entity;
++
++import net.minecraft.world.entity.animal.AbstractSchoolingFish;
++import org.bukkit.craftbukkit.CraftServer;
++import org.bukkit.craftbukkit.entity.CraftFish;
++
++public class PaperSchoolingFish extends CraftFish implements SchoolableFish {
```

Comment:
Since a new non-interface has been added here to the whole CraftEntity hierarchy, I think it should be added to the massive method in CraftEntity for the creation of new entities.
![image](https://user-images.githubusercontent.com/15055071/147781908-4f62a982-b9d5-4528-bda3-0390e6a753b1.png)

Reply:
Do I need to update 
![image](https://user-images.githubusercontent.com/23108066/147785284-75317f29-8012-4235-bd02-9e59b636348f.png)
as well?

Reply:
yeah, I suppose it should be changed there too.